lib/ansible/modules/network/iosxr/iosxr_banner.py

```diff
@@ -79,60 +79,140 @@
+        self._module = module
+        self._result = {'changed': False, 'warnings': warnings}
+        self._want = {}
+        self._have = {}
+
+    def map_params_to_obj(self):
+        text = self._module.params['text']
+        if text:
+            text = "{!r}".format(str(text).strip())
+        self._want.update({
+            'banner': self._module.params['banner'],
+            'text': text,
+            'state': self._module.params['state']
+        })
+
+
+class CliConfiguration(ConfigBase):
+    def __init__(self, module, warnings):
+        super(CliConfiguration, self).__init__(module, warnings)
+
+    def map_obj_to_commands(self):
+        commands = list()
+        state = self._module.params['state']
+        if state == 'absent':
+            if self._have.get('state') != 'absent' and ('text' in self._have.keys() and self._have['text']):
+                commands.append('no banner {!s}'.format(self._module.params['banner']))
+        elif state == 'present':
+            if (self._want['text'] and
+                    self._want['text'].encode().decode('unicode_escape').strip("'") != self._have.get('text')):
+                banner_cmd = 'banner {!s} '.format(self._module.params['banner'])
+                banner_cmd += self._want['text'].strip()
+                commands.append(banner_cmd)
+        self._result['commands'] = commands
+        if commands:
+            if not self._module.check_mode:
+                load_config(self._module, commands, self._result['warnings'], commit=True)
+            self._result['changed'] = True
+
+    def map_config_to_obj(self):
+        cli_filter = 'banner {!s}'.format(self._module.params['banner'])
+        output = get_config(self._module, config_filter=cli_filter)
+        match = re.search(r'banner (\S+) (.*)', output, re.DOTALL)
+        if match:
+            text = match.group(2).strip("'")
+        else:
+            text = None
+        obj = {'banner': self._module.params['banner'], 'state': 'absent'}
+        if output:
+            obj['text'] = text
+            obj['state'] = 'present'
+        self._have.update(obj)
+
+    def run(self):
+        self.map_params_to_obj()
+        self.map_config_to_obj()
+        self.map_obj_to_commands()
+
+        return self._result
+
+
+class NCConfiguration(ConfigBase):
+    def __init__(self, module, warnings):
+        super(NCConfiguration, self).__init__(module, warnings)
+        self._banners_meta = collections.OrderedDict()
+        self._banners_meta.update([
+            ('banner', {'xpath': 'banners/banner', 'tag': True, 'attrib': "operation"}),
+            ('a:banner', {'xpath': 'banner/banner-name'}),
+            ('a:text', {'xpath': 'banner/banner-text', 'operation': 'edit'})
+        ])
+
+    def map_obj_to_commands(self):
+        state = self._module.params['state']
+        _get_filter = build_xml('banners', xmap=self._banners_meta, params=self._module.params, opcode="filter")
```

Comment:
I think it is better to pass module as argument to build_xml() and handle lxml install check-in module_utils instead of having it in each module

Reply:
Module object is of no use in build_xml. instead moved out lib install check in module.

Reply:
Having this check in each module will result in code duplication. `module` param will be used if the check is moved to utils function

Reply:
build_xml may not have access to the `module` always (like from plugins); so moving the install check to utils for modules and plugins will do a separate install check.